### ISSUE TITLE
Had to clear variables twice

### ISSUE THREAD

**Issue by vitorpbarbosa7** [spyder-ide/spyder#11757]
### What steps will reproduce the problem?

Run any code with several variables (15 for example)

Select then all and press delete. 

It will delete some and reappear others. 

So, to really delete all variables, you have to select all of them again and press delete again. 

### Traceback
```python-traceback
Exception in comms call remove_value:
  File "C:\Users\vitor\Anaconda3\lib\site-packages\spyder_kernels\comms\commbase.py", line 342, in _handle_remote_call
    buffer['call_kwargs'])
  File "C:\Users\vitor\Anaconda3\lib\site-packages\spyder_kernels\comms\commbase.py", line 353, in _remote_callback
    *call_args, **call_kwargs)
  File "C:\Users\vitor\Anaconda3\lib\site-packages\spyder_kernels\console\kernel.py", line 234, in remove_value
    ns.pop(name)
KeyError: 'classe'
Exception in comms call remove_value:
  File "C:\Users\vitor\Anaconda3\lib\site-packages\spyder_kernels\comms\commbase.py", line 342, in _handle_remote_call
    buffer['call_kwargs'])
  File "C:\Users\vitor\Anaconda3\lib\site-packages\spyder_kernels\comms\commbase.py", line 353, in _remote_callback
    *call_args, **call_kwargs)
  File "C:\Users\vitor\Anaconda3\lib\site-packages\spyder_kernels\console\kernel.py", line 234, in remove_value
    ns.pop(name)
KeyError: 'labelencoder_previsores'
Exception in comms call remove_value:
  File "C:\Users\vitor\Anaconda3\lib\site-packages\spyder_kernels\comms\commbase.py", line 342, in _handle_remote_call
    buffer['call_kwargs'])
  File "C:\Users\vitor\Anaconda3\lib\site-packages\spyder_kernels\comms\commbase.py", line 353, in _remote_callback
    *call_args, **call_kwargs)
  File "C:\Users\vitor\Anaconda3\lib\site-packages\spyder_kernels\console\kernel.py", line 234, in remove_value
    ns.pop(name)
KeyError: 'scaler'
Exception in comms call remove_value:
  File "C:\Users\vitor\Anaconda3\lib\site-packages\spyder_kernels\comms\commbase.py", line 342, in _handle_remote_call
    buffer['call_kwargs'])
  File "C:\Users\vitor\Anaconda3\lib\site-packages\spyder_kernels\comms\commbase.py", line 353, in _remote_callback
    *call_args, **call_kwargs)
  File "C:\Users\vitor\Anaconda3\lib\site-packages\spyder_kernels\console\kernel.py", line 234, in remove_value
    ns.pop(name)
KeyError: 'previsores_treinamento'
Exception in comms call remove_value:
  File "C:\Users\vitor\Anaconda3\lib\site-packages\spyder_kernels\comms\commbase.py", line 342, in _handle_remote_call
    buffer['call_kwargs'])
  File "C:\Users\vitor\Anaconda3\lib\site-packages\spyder_kernels\comms\commbase.py", line 353, in _remote_callback
    *call_args, **call_kwargs)
  File "C:\Users\vitor\Anaconda3\lib\site-packages\spyder_kernels\console\kernel.py", line 234, in remove_value
    ns.pop(name)
KeyError: 'previsores_teste'
Exception in comms call remove_value:
  File "C:\Users\vitor\Anaconda3\lib\site-packages\spyder_kernels\comms\commbase.py", line 342, in _handle_remote_call
    buffer['call_kwargs'])
  File "C:\Users\vitor\Anaconda3\lib\site-packages\spyder_kernels\comms\commbase.py", line 353, in _remote_callback
    *call_args, **call_kwargs)
  File "C:\Users\vitor\Anaconda3\lib\site-packages\spyder_kernels\console\kernel.py", line 234, in remove_value
    ns.pop(name)
KeyError: 'classe_treinamento'
Exception in comms call remove_value:
  File "C:\Users\vitor\Anaconda3\lib\site-packages\spyder_kernels\comms\commbase.py", line 342, in _handle_remote_call
    buffer['call_kwargs'])
  File "C:\Users\vitor\Anaconda3\lib\site-packages\spyder_kernels\comms\commbase.py", line 353, in _remote_callback
    *call_args, **call_kwargs)
  File "C:\Users\vitor\Anaconda3\lib\site-packages\spyder_kernels\console\kernel.py", line 234, in remove_value
    ns.pop(name)
KeyError: 'classe_teste'
Exception in comms call remove_value:
  File "C:\Users\vitor\Anaconda3\lib\site-packages\spyder_kernels\comms\commbase.py", line 342, in _handle_remote_call
    buffer['call_kwargs'])
  File "C:\Users\vitor\Anaconda3\lib\site-packages\spyder_kernels\comms\commbase.py", line 353, in _remote_callback
    *call_args, **call_kwargs)
  File "C:\Users\vitor\Anaconda3\lib\site-packages\spyder_kernels\console\kernel.py", line 234, in remove_value
    ns.pop(name)
KeyError: 'lda'
```

## Versions

* Spyder version: 4.0.1 
* Python version: 3.7.6
* Qt version: 5.9.6
* PyQt5 version: 5.9.2
* Operating System: Windows 10

### Dependencies

```
atomicwrites >=1.2.0         :  1.3.0 (OK)
chardet >=2.0.0              :  3.0.4 (OK)
cloudpickle >=0.5.0          :  1.3.0 (OK)
diff_match_patch >=20181111  :  20181111 (OK)
intervaltree                 :  None (OK)
IPython >=4.0                :  7.12.0 (OK)
jedi =0.14.1                 :  0.14.1 (OK)
nbconvert >=4.0              :  5.6.1 (OK)
numpydoc >=0.6.0             :  0.9.2 (OK)
pexpect >=4.4.0              :  4.8.0 (OK)
pickleshare >=0.4            :  0.7.5 (OK)
psutil >=0.3                 :  5.6.7 (OK)
pygments >=2.0               :  2.5.2 (OK)
pylint >=0.25                :  2.4.4 (OK)
pyls >=0.31.2;<0.32.0        :  0.31.7 (OK)
zmq >=17                     :  18.1.1 (OK)
qdarkstyle >=2.7             :  2.8 (OK)
qtawesome >=0.5.7            :  0.6.1 (OK)
qtconsole >=4.6.0            :  4.6.0 (OK)
qtpy >=1.5.0                 :  1.9.0 (OK)
rtree >=0.8.3                :  0.9.3 (OK)
sphinx >=0.6.6               :  2.4.0 (OK)
spyder_kernels >=1.8.1;<2.0.0:  1.8.1 (OK)
watchdog                     :  None (OK)
cython >=0.21                :  0.29.15 (OK)
matplotlib >=2.0.0           :  3.1.3 (OK)
numpy >=1.7                  :  1.18.1 (OK)
pandas >=0.13.1              :  1.0.1 (OK)
scipy >=0.17.0               :  1.4.1 (OK)
sympy >=0.7.3                :  1.5.1 (OK)
```

**Comment by ccordoba12** [comment]
Duplicate of #11001

**Comment by ccordoba12** [comment]
This will be fixed in our next version (4.1.0), to be released at the end of the week.
